### ISSUE TITLE
[Proposal] Helper function for conditionally joining class names together

### ISSUE THREAD

**Issue by obaydmir** [laravel/ideas#1824]
This feature request is inspired by a package named [classnames](https://github.com/JedWatson/classnames) for JavaScript maintained by Jed Watson. It would be helpful for laravel to also have a helper like this so conditionally rendering class names becomes less verbose.

This helper returns a string containing all class names which do not have a condition set as element value or have resulted in true boolean value.

**The basic implementation of this helper would be:**

```
if (! function_exists('cn')) {
    /**
     * Conditional return classnames.
     *
     * @param array $list
     * @return string
     */
    function cn(array $list) {

        $renderList = [];
        foreach ($list as $key => $value) {
            if (is_int($key)) {
                $renderList[] = $value;
            } elseif ($value) {
                $renderList[] = $key;
            }
        }

        return implode(' ', $renderList);
    }
}
```

**Example usage:**

```
// > "default-class class-a"
$classnames = cn([
    'default-class',
    'class-a' => true,
    'class-b' => false,
]);
```

**Comment by svenluijten** [comment]
You can create your own `functions.php` file an include it in your project via [Composer's autoloading `files` option](https://github.com/laravel/framework/blob/d9b24caa67eacd22d917a5fd3b3601937112a329/composer.json#L94-L97), I don't think this `cn` function would fit in the core framework.

**Comment by garstas** [comment]
@obaydmir , can you provide a real-life use-case for this feature, please? I'm intrigued :thinking:

**Comment by obaydmir** [comment]
@svenluijten I already have, but thanks for the help.

The reason why I think it would help is because I now have to put the logic (if, else...) in my template. It sometimes becomes really verbose. But I also repeat the same pattern.

**Comment by stefanfisk** [comment]
When there’s a bunch of classes that you want to conditionally toggle, something like `classnames` really helps keeping templates concise compared to if-ing parts of a string. I’ve got a custom helper for this that I use extensively in all my project.

**Comment by obaydmir** [comment]
@stefanfisk Exactly. I'm using this helper function for those kinds of purposes.
